src/pybind/mgr/cephadm/serve.py

```diff
@@ -1004,8 +1004,15 @@ def _check_daemons(self) -> None:
 
         # do daemon post actions
         for daemon_type, daemon_descs in daemons_post.items():
-            if daemon_type in self.mgr.requires_post_actions:
-                self.mgr.requires_post_actions.remove(daemon_type)
+            run_post = False
+
+            x = ''
+
+            for d in daemon_descs:
+                if d.name() in self.mgr.requires_post_actions:
+                    self.mgr.requires_post_actions.remove(d.name())
```

Comment:
what if the post action has a transient failure? remove the daemon name iff the post action succeeds?

Reply:
how do we deal with non-transient errors then?

Reply:
maybe a thing for another PR?